shared/helpers/puppeteer.js

```diff
@@ -95,7 +95,11 @@ async function renderAndInitialise(page, componentName, options) {
   // as running it inside the body of the next `evaluate`
   // didn't provide a reliable execution
   if (options.beforeInitialisation) {
-    await page.$eval('[data-module]', options.beforeInitialisation)
+    await page.$eval(
+      '[data-module]',
+      options.beforeInitialisation,
+      componentName
```

Comment:
@romaricpascal Would you mind using what we call `moduleName`

Each class has a static property like `Accordion.moduleName` as `'govuk-accordion'`

But it can be formed from `govuk-${componentName}` to maintain compatibility with `initAll()`

```suggestion
      `govuk-${componentName}`
```

Reply:
Actually, no need for this extra param, `$module` already has al the information we need: we can `.getAttribute('data-module')` (not `.dataset.module` because of typing), and that gets us the value we want when replacing with a dummy element. 

Makes the replacement a little clearer as it shows the attribute value came from the original element, and also avoid having the template string in both places 😊

Reply:
That's fine too. Can you check you're happy keeping the loose `[data-module]` selector in https://github.com/alphagov/govuk-frontend/pull/4104#pullrequestreview-1617358822?

Was hoping to prevent other tests from selecting the wrong module:

### Before
```mjs
// ✅ <svg data-module="colin"></svg>
await page.$eval('[data-module]', options.beforeInitialisation)
```

### After
```mjs
// ❌ <svg data-module="colin"></svg>
await page.$eval(`[data-module="${moduleName}"]`, options.beforeInitialisation)
```

Reply:
I'd say we're fine for that PR: `renderAndInitialise` only has one component on the page, so the topmost element with `[data-module]` is what we're after.

If we were to use the `moduleName` (which I'm not against, just think it's not in the scope of this PR), I'd rather have that in its own PR, so we can rework the helper a little and:
- grab that `moduleName` directly from the component class passed to the helper rather than re-implement that it's `govuk-${componentName}`
- take advantage of that to change the following `evaluate` into `$eval` as well, as it's no longer necessary to do `document.querySelector` (an opportunity missed in https://github.com/alphagov/govuk-frontend/commit/b5528761a799967b4d8aaa7fcab12e9e94cc6855)

Reply:
Brill. Aware it was unanswered until now, so I'm happy if you're happy